app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryInit.kt

```diff
@@ -1,15 +1,27 @@
 package org.simple.clinic.drugs.selection.custom
 
 import com.spotify.mobius.First
+import com.spotify.mobius.First.first
 import com.spotify.mobius.Init
-import org.simple.clinic.mobius.first
+import org.simple.clinic.drugs.selection.custom.OpenAs.New.FromDrugList
+import org.simple.clinic.drugs.selection.custom.OpenAs.New.FromDrugName
+import org.simple.clinic.drugs.selection.custom.OpenAs.Update
 
 class CustomDrugEntryInit : Init<CustomDrugEntryModel, CustomDrugEntryEffect> {
   override fun init(model: CustomDrugEntryModel): First<CustomDrugEntryModel, CustomDrugEntryEffect> {
-    return when (model.openAs) {
-      is OpenAs.New.FromDrugList -> first(model.drugInfoProgressStateLoading(), FetchDrug(model.openAs.drugUuid), LoadDrugFrequencyChoiceItems)
-      is OpenAs.New.FromDrugName -> first(model.drugNameLoaded(model.openAs.drugName).drugInfoProgressStateLoaded(), LoadDrugFrequencyChoiceItems)
-      is OpenAs.Update -> first(model.drugInfoProgressStateLoading(), FetchPrescription(model.openAs.prescribedDrugUuid), LoadDrugFrequencyChoiceItems)
+    val updatedModel = when (model.openAs) {
+      is FromDrugList, is Update -> model.drugInfoProgressStateLoading()
+      is FromDrugName -> model.drugNameLoaded(model.openAs.drugName).drugInfoProgressStateLoaded()
     }
+
+    val effect = when (updatedModel.openAs) {
+      is FromDrugList -> FetchDrug(updatedModel.openAs.drugUuid)
+      is Update -> FetchPrescription(updatedModel.openAs.prescribedDrugUuid)
+      is FromDrugName -> null
+    }
+
+    val effects = mutableSetOf(LoadDrugFrequencyChoiceItems, effect)
```

Comment:
It does give an android warning tho

Reply:
Aren't we trying to avoid those?

Reply:
This should be fine here, we are not trying to be exhaustive here since we are adding to the effects list, so we don't need to handle all the `OpenAs` types in the `when` statement.